Use segmented control for mask selections

jy2wong opened the pull request.

blanchardjeremy commented:
@jy2wong can you rebase your new commits that are unique to the PR on `main` now that the masks update is live there?

beshaya commented:
Found a formatting oddity at around 1200px wide:
![image](https://user-images.githubusercontent.com/5877421/107103455-8a899e80-67d2-11eb-85c3-2232552f45dd.png)

And again around 720px:
![image](https://user-images.githubusercontent.com/5877421/107103481-a2612280-67d2-11eb-88ee-a2efd403b2c4.png)

"Hover/Tap for descriptions" instruction probably needs to be changed depending on layout (hover doesn't work at mobile layout, tap doesn't work for desktop layout). Possibly this instruction can just be removed; the behavior feels quite intuitive and responsive to me.

I quite like how the description of the chosen mask is visible in mobile layout, maybe add something like that to desktop?

jy2wong commented:
> Found a formatting oddity at around 1200px wide:
> ![image](https://user-images.githubusercontent.com/5877421/107103455-8a899e80-67d2-11eb-85c3-2232552f45dd.png)
> 
fixed (I think?)

> And again around 720px:
> ![image](https://user-images.githubusercontent.com/5877421/107103481-a2612280-67d2-11eb-88ee-a2efd403b2c4.png)
> 
fixed

> "Hover/Tap for descriptions" instruction probably needs to be changed depending on layout (hover doesn't work at mobile layout, tap doesn't work for desktop layout). Possibly this instruction can just be removed; the behavior feels quite intuitive and responsive to me.
> 

> I quite like how the description of the chosen mask is visible in mobile layout, maybe add something like that to desktop?
done

jy2wong commented:
This is ready for code review now.

jy2wong commented on src/components/calculator/controls/SegmentedControl.tsx:
```diff
@@ -0,0 +1,120 @@
+  const activeValue = typeof dataValue === 'string' ? dataValue : ''
+  const { t } = useTranslation()
+  // 0.25 -> "1/4x"
+  function formatRiskMultiplierInternal(multiplier: number): string {
```
This one is different from the one in SelectControl.tsx. The select one does 0.2 -> [1/5th the risk] whereas this one does 0.2 -> 1/5x.

jy2wong commented on src/components/calculator/ActivityRiskControls.tsx:
```diff
@@ -61,26 +63,77 @@ export const ActivityRiskControls: React.FunctionComponent<{
         </div>
       )}
-      <SelectControl
+      <SegmentedControl
```
done

jy2wong replied:
done

beshaya reviewed (approved):
lgtm
